protocols/imap/src/main/java/org/apache/james/imap/processor/AbstractMailboxProcessor.java

```diff
@@ -245,37 +246,38 @@ private void addFlagsResponses(ImapSession session,
         while (it.hasNext()) {
             MessageResult mr = it.next();
             final MessageUid uid = mr.getUid();
-            int msn = selected.msn(uid);
-            if (msn == SelectedMailbox.NO_SUCH_MESSAGE) {
+            selected.msn(uid).fold(() -> {
```

Comment:
Method extraction?

Reply:
the whole method is about what happens here. I tried but failed to extract that code in a meaningful way. BTW, this code is not related to the change I did.